src/main/java/fi/helsinki/cs/tmc/core/domain/submission/AdaptiveSubmissionResult.java

```diff
@@ -1,32 +1,34 @@
 package fi.helsinki.cs.tmc.core.domain.submission;
 
-import com.google.gson.annotations.SerializedName;
+import fi.helsinki.cs.tmc.langs.domain.TestResult;
 
+import java.io.Serializable;
 import java.util.ArrayList;
 
-public class AdaptiveSubmissionResult {
-    public AdaptiveSubmissionResult() {
-        status = SubmissionResult.Status.ERROR;
-        error = null;
-        points = 0;
-    }
-
-    @SerializedName("error")
-    private final String error;
+public class AdaptiveSubmissionResult implements Serializable {
 
-    @SerializedName("status")
-    private final SubmissionResult.Status status;
+    public SubmissionResult.Status status;
+    public ArrayList<TestResult> testResults = new ArrayList<>();
+    public String error;
 
-    @SerializedName("points")
-    private final int points;
+    public AdaptiveSubmissionResult() {
+    }
 
     public SubmissionResult toSubmissionResult() {
         SubmissionResult submissionResult = new SubmissionResult();
+        if (error == null && status == SubmissionResult.Status.ERROR) {
+            error = "errored";
+        }
         submissionResult.setError(error);
         submissionResult.setStatus(status);
         ArrayList<String> submissionPoints = new ArrayList<>();
-        submissionPoints.add(String.valueOf(points));
+        if (status == SubmissionResult.Status.OK) {
+            submissionPoints.add(String.valueOf("1"));
```

Comment:
Doesn't this (and the similar line in the `else` block) just end up being `String.valueOf(((Object)"1").toString())`? Is there some specific reason you'd want to cast a `String` to an `Object` and then back to a `String` rather than directly adding the `String`?